### DISCUSSION TITLE
Does JSqlParser support Cloudant

### DISCUSSION THREAD

**Discussion by ashishramtri** [JSQLParser/JSqlParser#1499] [Q&A]
I am working with the Cloudant data source. And currently in the process of creating a SQL interface for an end-user, wherein I am planning on using JSQLParser to get all the visitors' hierarchy.  And implement the visitor pattern so that for every keyword, we can create a Cloudant Query. And this JSON/Cloudant query will then be fired onto the Cloudant database.

Will JSQLParser come in handy in this case?

**Comment by manticore-projects** [comment]
Greetings.

Do you mean, you would like to translate any SQL Statement into the Cloudant compatible JSON request?
If so, then yes, JSQLParser would be the perfect translator. You could start with the `Deparser` implementation and rewrite all the methods in order to get the JSON request instead of the SQL string.

Good luck!

PS: as a starting point, you may look at https://github.com/manticore-projects/MJdbcUtils where I rewrite SQL statements in a similar way.

**Reply by ashishramtri** [reply]
That's absolutely right, sir!

The plan is to:
-Take SQL as an input from the end-user and create an Abstract Syntax Tree using the JSqlParser
-Implement the visitor pattern which would help to translate the SQL statement into the Cloudant Query (?)
-And fire this Cloudant Query onto the Cloudant data source
-And especially, convert the JSON results format into tabular ResultSet format

I am totally new to working with parsers. Hence, would REALLY appreciate any other approach/resource that you could redirect me towards from your experience.

Thanks for redirecting me to the above-mentioned resource(as a starting point) though! 👍

**Comment by manticore-projects** [comment]
Sounds all feasible and good.
Let me know any questions your have on step 2) "Implement the visitor pattern which would help to translate the SQL statement into the Cloudant Query".

Good luck!

**Reply by ashishramtri** [reply]
Sure, will do.

Thank you!!!
